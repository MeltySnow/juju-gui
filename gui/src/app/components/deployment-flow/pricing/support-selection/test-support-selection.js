/* Copyright (C) 2018 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const Analytics = require('../../../../../test/fake-analytics');
const DeploymentSupportSelection = require('./support-selection');

describe('DeploymentSupportSelection', () => {

  const renderComponent = (options = {}) => enzyme.shallow(
    <DeploymentSupportSelection
      analytics={Analytics}
      getSLAMachineRates={options.getSLAMachineRates || sinon.stub()}
      machineCount="3"
      setSLA={options.setSLA || sinon.stub()} />
  );

  it('can render', () => {
    const data = {
      advanced: '0.110',
      essential: '0.011',
      standard: '0.055',
      unsupported: '0.000'
    };
    const wrapper = renderComponent({
      getSLAMachineRates: cb => cb(data)
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('shows the spinner while requesting SLA rates', () => {
    const wrapper = renderComponent();
    assert.equal(wrapper.find('Spinner').length, 1);
  });

  it('can select a plan', () => {
    const data = {
      advanced: '0.110',
      essential: '0.011',
      standard: '0.055',
      unsupported: '0.000'
    };
    const setSLA = sinon.stub();
    const wrapper = renderComponent({
      getSLAMachineRates: cb => cb(data),
      setSLA: setSLA
    });
    const instance = wrapper.instance();
    const plan = wrapper.find('DeploymentSupportSelectionPlan').at(1);
    assert.equal(plan.prop('selected'), false);
    plan.props().onSelect();
    assert.equal(instance.state.selectedPlan, 'standard');
    wrapper.update();
    assert.equal(
      wrapper.find('DeploymentSupportSelectionPlan').at(1).prop('selected'),
      true);
    assert.deepEqual(setSLA.args[0], [{name: 'standard', hourPrice: '0.055'}]);
  });
});
