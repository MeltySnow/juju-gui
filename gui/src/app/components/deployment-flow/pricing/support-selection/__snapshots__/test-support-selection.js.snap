// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentSupportSelection can render 1`] = `
<div
  className="deployment-support-selection equal-height"
>
  <DeploymentSupportSelectionPlan
    analytics={
      Object {
        "addCategory": [Function],
        "sendEvent": [Function],
      }
    }
    classes={Array []}
    features={
      Array [
        "8hx5d ticked",
      ]
    }
    hourPrice="0.011"
    key="Essential"
    machineCount={3}
    onSelect={[Function]}
    selected={false}
    title="Essential"
  />
  <DeploymentSupportSelectionPlan
    analytics={
      Object {
        "addCategory": [Function],
        "sendEvent": [Function],
      }
    }
    classes={Array []}
    features={
      Array [
        "10x5 phone support",
        "2hr critical response",
      ]
    }
    hourPrice="0.055"
    key="Standard"
    machineCount={3}
    onSelect={[Function]}
    selected={false}
    title="Standard"
  />
  <DeploymentSupportSelectionPlan
    analytics={
      Object {
        "addCategory": [Function],
        "sendEvent": [Function],
      }
    }
    classes={
      Array [
        "last-col",
      ]
    }
    features={
      Array [
        "24x7 phone support",
        "1hr critical response",
      ]
    }
    hourPrice="0.110"
    key="Advanced"
    machineCount={3}
    onSelect={[Function]}
    selected={false}
    title="Advanced"
  />
</div>
`;
